libfreerdp/locale/locale.c

```diff
@@ -850,13 +850,23 @@ int freerdp_detect_keyboard_layout_from_system_locale(DWORD* keyboardLayoutId)
 	freerdp_get_system_language_and_country_codes(language, ARRAYSIZE(language), country,
 	                                              ARRAYSIZE(country));
 
-	if ((strcmp(language, "C") == 0) || (strcmp(language, "POSIX") == 0))
+	char locale[LOCALE_COUNTRY_LEN + LOCALE_LANGUAGE_LEN + 2] = { 0 };
+	_snprintf(locale, sizeof(locale), "%s_%s", language, country);
+	return freerdp_detect_keyboard_layout_from_locale(locale, keyboardLayoutId);
+}
+
+int freerdp_detect_keyboard_layout_from_locale(const char* localestr, DWORD* pKeyboardLayoutId)
```

Comment:
Can't we return the layout id directly as a result instead of passing an out parameter i.e. `DWORD freerdp_detect_keyboard_layout_from_locale(const char* localestr)`?
Makes the function more comfortable to use.